src/lua/LuaUtils.cpp

```diff
@@ -303,6 +303,22 @@ static int l_readonly_table_ipairs(lua_State *l)
 	return 3;
 }
 
+void pi_lua_push_date_time(lua_State *l, const Time::DateTime &dt)
```

Comment:
Note that this could also be exposed as a `pi_lua_generic_push` overload, allowing the use of `DateTime` objects via the `LuaPush<T>()` mechanism. See `LuaPushPull.h` for details on the specific API arguments.

The `LuaPush/Pull` API is used in quite a few places to provide C++ template "glue" for a nicer API when calling or moving data into Lua and vice versa.

Reply:
I did the bare minimum as my knowledge of Lua/C++ interoperability is a work-in-progress, and my C++ skills also aren't the best (I mainly program in C).

I'll take a look and improve my PR. Thank you for the suggestion.

Reply:
No worries - this comment was mostly aimed at providing you with additional knowledge regarding our specific flavor of C++/Lua interop. The `pi_lua_generic_push` overload can be implemented on an as-needed basis in the future, it's definitely not a requirement for merge.

Reply:
No I'd very much like to use the overloaded "push pull" approach, just not sure how as the other implementations are primitive types. Is it ok to just add a new table to the lua state for compound types in the push-pull overloads?

Reply:
Fixed (?) in e3fca64b76ea7b51d5f18b7e63a2c7d6cf486a55

Reply:
Didn't get a chance to answer this last night - the API contract of `pi_lua_generic_push` expects that a single value is left on the top of the Lua stack after the function has finished. That value can be a primitive type, or a compound type like a table or userdata - the important thing is just to not leave any "extra values" on the stack or the calling code's assumptions will break.

Reply:
... eh, no idea if what I did does that; as I said, I have literally zero idea (well, maybe 0.1% now) of how Lua and C(++) interact.  That said, initial testing appears to show things working as expected following this rework..